interface/repository/csv_item_repository.go

```diff
@@ -0,0 +1,62 @@
+package repository
+
+import (
+	"academy_bootcamp/application/contracts"
+	"academy_bootcamp/domain/model"
+	"academy_bootcamp/infrastructure/datastore"
+	"errors"
+	"io"
+	"strconv"
+)
+
+type CSVItemRepository struct {
+	Handler *datastore.CSVHandler
+}
+
+func NewItemRepository(h *datastore.CSVHandler) contracts.ItemRepository {
+	return &CSVItemRepository{h}
+}
+
+func (r *CSVItemRepository) FindByID(id uint64) (*model.Item, error) {
+	var rec *model.Item
+
+	erro := r.Handler.BuildHandler()
+
+	if erro != nil {
+		return nil, erro
+	}
+	for {
+
+		record, err := r.Handler.Read()
+		if err == io.EOF {
+			break
+		}
```

Comment:
Why name it `erro` instead of the well-known `err` name?

No need to leave blank lines after the for.

Cuddle the assignation and checking of your errors.

```suggestion
func (r *CSVItemRepository) FindByID(id uint64) (*model.Item, error) {
	var rec *model.Item

	err := r.Handler.BuildHandler()
	if err != nil {
		return nil, err
	}
	for {
		record, err := r.Handler.Read()
		if err == io.EOF {
			break
		}
```